src/Functional.CQS.AOP.IoC.SimpleInjector.Caching/ConventionBasedDecoratorRegistrationGatewayExtensions.cs

```diff
@@ -0,0 +1,91 @@
+		/// <param name="configurationParameters">The configuration parameters.</param>
+		/// <returns></returns>
+		public static ConventionBasedDecoratorRegistrationGateway WithCachingDecorator<TCache>(this ConventionBasedDecoratorRegistrationGateway gateway, Func<TCache> cacheFactory, CachingModuleConfigurationParameters configurationParameters)
+			where TCache : class, IFunctionalCache
+		{
+			return gateway.WithCachingDecorator<TCache, NullCacheHitAndMissLogger, NullCacheItemReplacementLogger, NullCacheItemInvalidationLogger, NullCacheExceptionLogger>(cacheFactory, configurationParameters);
+		}
+
+		/// <summary>
+		/// Register all core components required for applying caching decorators to <see cref="IQueryHandler{TQuery, TResult}"/> and <see cref="IAsyncQueryHandler{TQuery, TResult}"/> implementations.
+		/// Only handlers with corresponding <see cref="IQueryResultCachingStrategy{TQuery, TResult}"/> implementations will have the caching decorator applied to them.
+		/// </summary>
+		/// <typeparam name="TCache">The cache type.  Will be registered as singleton.</typeparam>
+		/// <typeparam name="TCacheHitAndMissLogger">The logger for cache hits and misses.  Will be registered as singleton.</typeparam>
+		/// <typeparam name="TCacheItemReplacementOperationLogger">The logger for cache item replacement operations.  Will be registered as singleton.</typeparam>
+		/// <typeparam name="TCacheItemInvalidationOperationLogger">The logger for cache item invalidation operations.  Will be registered as singleton.</typeparam>
+		/// <typeparam name="TCacheExceptionLogger">The logger for cache exceptions.  Will be registered as singleton.</typeparam>
+		/// <param name="gateway">The gateway.</param>
+		/// <param name="cacheFactory">The factory method that will be invoked once to create a singleton instance of <typeparamref name="TCache"/>.</param>
+		/// <param name="configurationParameters">The configuration parameters.</param>
+		public static ConventionBasedDecoratorRegistrationGateway WithCachingDecorator<TCache, TCacheHitAndMissLogger, TCacheItemReplacementOperationLogger, TCacheItemInvalidationOperationLogger, TCacheExceptionLogger>(this ConventionBasedDecoratorRegistrationGateway gateway, Func<TCache> cacheFactory, CachingModuleConfigurationParameters configurationParameters)
+			where TCache : class, IFunctionalCache
+			where TCacheHitAndMissLogger : class, ILogFunctionalCacheHitsAndMisses
+			where TCacheItemReplacementOperationLogger : class, ILogFunctionalCacheItemReplacementOperations
+			where TCacheItemInvalidationOperationLogger : class, ILogFunctionalCacheItemInvalidationOperations
+			where TCacheExceptionLogger : class, ILogFunctionalCacheExceptions
+		{
+			if (cacheFactory == null) throw new ArgumentNullException(nameof(cacheFactory));
+			if (configurationParameters == null) throw new ArgumentNullException(nameof(configurationParameters));
+
+			gateway.Container.RegisterSingleton<IFunctionalCache>(cacheFactory);
```

Comment:
This block of code deals with components that are dependencies of the decorators, but are not the decorators themselves.  These dependencies _could_ be used without the decorators, so will register them regardless so things do not break at runtime if people decide to turn caching decorations off.